arrows/core/derive_metadata.cxx

```diff
@@ -52,6 +52,13 @@ get_platform_rotation( kwiver::vital::metadata_sptr const& metadata )
   auto const pitch = pitch_item.as_double() * kv::deg_to_rad;
   auto const roll = roll_item.as_double() * kv::deg_to_rad;
 
+  if( pitch < -20 * kv::deg_to_rad || pitch > 20 * kv::deg_to_rad ||
```

Comment:
±20° pitch seems somewhat small. I wonder if we really want to limit this code to values that are likely based on expected platforms? Or should we limit it to values that are "clearly" wrong, i.e. ±180°? (Is there any chance this code would get used on footage from, say, an acrobatic quadcopter?)

Reply:
As below, I think this code is pretty questionable but it's worth faithfully recreating it for now. In future work, it seems like it would be fairly simple to just project the image corners onto the ground plane and then work from there, with a few different cases.

Reply:
What does projecting the corners have to do with validating the platform orientation?

> I think this code is pretty questionable but it's worth faithfully recreating it for now.

For *algorithmic* stuff, I'd agree. For input validation, I'm less inclined to be blindly adherent, since the only difference would be that our implementation might try to give a result whereas theirs just rejects the input.

Reply:
I honestly wouldn't have expected it, but these constants actually have a reason behind them -- they are part of the [MISB standard](https://gwg.nga.mil/misb/docs/standards/ST0601.6.pdf): 

> Map -(2^15-1)..(2^15-1) to +/-20. Use -(2^15) as 'out of range' indicator. [Page 15]

MISB uses integers mapped to bounded numerical ranges, not floating point, to represent a lot of different values -- for the compactness, I guess. Thus, these numerical checks are actually only supposed to be a sort of "is not null" check, since MISB can't represent values outside the ranges.

Personally, I would suggest that these validity checks be moved to wherever the data is being read in. The GSD functionality should be independent of the source of its parameters, and should only check for invalidity with respect to its own operations -- for example, there is no GSD if the sensor is pointing at the sky. GSD is still a valid metric with a pitch of -21 degrees, so the GSD function should not fail just because the only data source we have right now happens to use -21 degrees as a null value. 

The MISB reader should do these MISB-specific checks and just not add the invalid entries to the metadata structure in the first place, unifying the error checking with the "missing data" checks our code already contains.